### PR TITLE
[libc] Remove more libc dependencies from the RPC header

### PR DESCRIPTION
**Pull request by jhuber6**
Summary:
The end goal is to make `rpc.h` a standalone header so that other
projects can include it without leaking `libc` internals. I'm trying to
replace stuff slowly before pulling it out all at once to reduce the
size of the changes.

This patch removes the atomic and a few sparse dependencies. Now we
mostly rely on the GPU utils, the sleep function, optional, and the
type traits. I'll clean these up in future patches. This removed the old
stuff I had around the memcpy, but I think that it's not quite as bad as
it once was, as it removes a branch and only uses a few extra VGPRs
since I believe the builtin memcpy was improved for AMD.

### CODE REVIEW

File: libc/src/__support/RPC/rpc_util.h [hunk 1]
```diff
@@ -10,20 +10,16 @@
...
-#include "src/string/memory_utils/generic/byte_per_byte.h"
-#include "src/string/memory_utils/inline_memcpy.h"
 
 namespace LIBC_NAMESPACE_DECL {
 namespace rpc {
 
 /// Conditional to indicate if this process is running on the GPU.
 LIBC_INLINE constexpr bool is_process_gpu() {
-#if defined(LIBC_TARGET_ARCH_IS_GPU)
+#if defined(__NVPTX__) || defined(__AMDGPU__)
```
Comment (michaelrj-google) [inline libc/src/__support/RPC/rpc_util.h:22]: would it make sense to have a `gpu_common.h` public header where you could define `LIBC_TARGET_ARCH_IS_GPU`?

**Review by michaelrj-google — APPROVED** [review]
LGTM from the libc side

File: libc/src/__support/RPC/rpc_util.h (hunk 1 above)
Comment (jhuber6) [reply libc/src/__support/RPC/rpc_util.h:22]: I think I'm going to cram all of this in `rpc_util.h`. Question is where I should put those two headers when I'm done such that they can be included by everyone.

Comment (michaelrj-google) [reply libc/src/__support/RPC/rpc_util.h:22]: I was assuming they were going in `include/gpu/rpc_util.h`

**Pull request merged**